### PR TITLE
Pyu/afp 263 make examples code view collapsible

### PR DESCRIPTION
**Pull request by pyupew**
- The code view is expandable and collapsible.
- The code view is not resizable by dragging
- The code will automatically collapse if the window width is too small. Whether the view was expanded or collapsed (before the window resizing) is stored in state, and restored when the window is resized to be wide enough
- Prettier reformatting the code when the code view is resized remains a TODO
- The collapse button is not ideally positioned

### CODE REVIEW

**Comment by Noviny** [comment]
image

I know it's going to overlap something, but this sticks out quite a long way. Could we imitate the positioning for the nav-next button? About 50/50 in and out of the collapsible container?

File: packages/website/src/components/page-templates/package-example.js [hunk 1]
```diff
@@ -58,63 +78,127 @@ const ExampleHeading = styled.h2`
...
+CodeViewButton.propTypes = {
+  isCodeViewExpanded: PropTypes.bool.isRequired,
+  handleClick: PropTypes.func.isRequired,
+};
+
+class PackageExample extends React.Component {
+  state = {
+    showCodeViewButton: true,
+    isCodeViewExpanded: true,
+    shouldExpandCodeWithWideWindow: true,
```
Comment (Noviny) [inline packages/website/src/components/page-templates/package-example.js]: Just for funs, we're using react 16.8 (the one with hooks) - it'd be nice to start writing some of these components using a hooks architecture for state management over a class component.

Happy to pair with you on changing this over if you want to. (I also need to gain experience using hooks)

File: packages/website/src/components/page-templates/package-example.js [hunk 2]
```diff
@@ -58,63 +78,127 @@ const ExampleHeading = styled.h2`
...
-      </NavigationWrapper>
-    </>
+  handleClick = () => {
+    this.setState(state => ({
+      isCodeViewExpanded: !state.isCodeViewExpanded,
+      shouldExpandCodeWithWideWindow: !state.shouldExpandCodeWithWideWindow,
+    }));
+  };
+
+  handleResize = debounce(
```
Comment (Noviny) [inline packages/website/src/components/page-templates/package-example.js]: Debounce - excellent.

File: packages/website/src/components/page-templates/package-example.js [hunk 3]
```diff
@@ -58,63 +78,127 @@ const ExampleHeading = styled.h2`
...
+    </Button>
+  </div>
+);
 
-  & code {
-    font-family: Monaco, Menlo, monospace;
-  }
-`;
+CodeViewButton.propTypes = {
+  isCodeViewExpanded: PropTypes.bool.isRequired,
```
[the remark this answers is not in the document]
Comment (pyupew) [reply packages/website/src/components/page-templates/package-example.js]: yep the page templates just never got converted

File: packages/website/src/components/code-example/code-view.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  overflow-y: scroll;
+
+  &.collapsed {
+    width: 0;
+    overflow: hidden;
+  }
+`;
+
+const CodeStyle = styled.pre`
+  background-color: rgb(23, 43, 77);
```
[the remark this answers is not in the document]
Comment (pyupew) [reply packages/website/src/components/code-example/code-view.tsx]: It's the same background colour used by Atlaskit in its code sidebar. I was pulling the rgb value from inspecting the element in the browser, but have just discovered that it's just N800 so will change it

File: packages/website/src/components/page-templates/package-example.js (hunk 1 above)
Comment (pyupew) [reply packages/website/src/components/page-templates/package-example.js]: > No blocking changes, though I would be keen to do the hooks refactor as a fun Friday afternoon activity.

that sounds good

**Pull request merged**